react-native/src/TasksView.js

```diff
@@ -0,0 +1,190 @@
+import React, {useEffect, useState, useRef} from 'react';
+import Realm from 'realm';
+import {BSON} from 'realm';
+import {SafeAreaProvider} from 'react-native-safe-area-context';
+import {StyleSheet, Text, View} from 'react-native';
+import {Button, Overlay, ListItem} from 'react-native-elements';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import {appId} from '../realm';
+import {CreateToDoPrompt} from './CreateToDoPrompt';
+import TaskSchema from './TaskSchema';
+
+const app = Realm.App.getApp(appId);
+
+Icon.loadFont(); // load FontAwesome font
+
+export function TasksView({navigation}) {
+  const [tasks, setTasks] = useState([]);
+
+  // state value for toggable visibility of the 'CreateToDoPrompt' Model in the UI
+  const [createToDoOverlayVisible, setCreateToDoOverlayVisible] = useState(
+    false,
+  );
+
+  // Use a Ref to store the realm rather than the state because it is not
+  // directly rendered, so updating it should not trigger a re-render as using
+  // state would.
+  const realmReference = useRef(null);
+
+  useEffect(() => {
```

Comment:
I don't see the value in extracting this effect into its own hook. This effect is very specific to the Task View (in that it sets the Tasks to state and adds a listener on the Tasks. The realm itself is not stateful because it doesn't need to update the UI when it's changed.

 Originally I was thinking it could be a `useTasks` hook (sort of like in the tutorial that's currently in the docs - https://github.com/mongodb-university/realm-tutorial-react-native/blob/final/providers/TasksProvider.js, but I decided against it to keep the template app very simple.